src/plugins/intel_npu/src/plugin/npuw/partitioning/patterns/dcoff.cpp

```diff
@@ -329,6 +329,14 @@ bool DCOFFPassBase::matcher_callback(ov::pass::pattern::Matcher& m) {
     NPUW_ASSERT(ov::op::util::is_parameter(matched_nodeC));
 
     auto matched_paramA = std::static_pointer_cast<ov::op::v0::Parameter>(matched_nodeA);
+    // Transpose weights specifically for QWEN as of now.
+    if (getTransposeWeights()) {
+        ov::PartialShape current_shape = matched_paramA->get_partial_shape();
+        ov::Shape static_shape = current_shape.to_shape();
+        ov::Shape new_order = {static_shape[1], static_shape[2], static_shape[0]};
+        ov::PartialShape new_shape(new_order);
+        matched_paramA->set_partial_shape(new_shape);
+    }
     auto matched_valueB = std::static_pointer_cast<ov::op::v0::Constant>(matched_nodeB);
     auto matched_paramC = std::static_pointer_cast<ov::op::v0::Parameter>(matched_nodeC);
```

Comment:
So this only change is not enough.

If you reshape the weights here, you will also need to reshape the Scale coefficient parameters if Scale parameters stay in in the model (no DCOFF).

Reply:
Yes, reshaping both weights and scales - i.e.,  if reshapable.